lib/flutter_map_tappable_polyline.dart

```diff
@@ -208,7 +149,86 @@ class TappablePolylineLayer extends StatelessWidget {
     );
   }
 
-  double distance(Offset point1, Offset point2) {
+  void _handlePolylineTap(
+      TapUpDetails details, Function onTap, Function onMiss) {
+    var hit = false;
+
+    // We might hit close to multiple polylines. We will therefore keep a reference to these in this map.
+    Map<double, TaggedPolyline> candidates = {};
+
+    // Calculating taps in between points on the polyline. We
+    // iterate over all the segments in the polyline to find any
+    // matches with the tapped point within the
+    // pointerDistanceTolerance.
+    for (Polyline currentPolyline in polylineOpts.polylines) {
+      for (var j = 0; j < currentPolyline.offsets.length - 1; j++) {
+        // We consider the points point1, point2 and tap points in a triangle
+        var point1 = currentPolyline.offsets[j];
+        var point2 = currentPolyline.offsets[j + 1];
+        var tap = details.localPosition;
+
+        // To determine if we have tapped in between two po ints, we
+        // calculate the length from the tapped point to the line
+        // created by point1, point2. If this distance is shorter
+        // than the specified threshold, we have detected a tap
+        // between two points.
+        //
+        // We start by calculating the length of all the sides using pythagoras.
+        var a = _distance(point1, point2);
+        var b = _distance(point1, tap);
+        var c = _distance(point2, tap);
```

Comment:
We are introducing breaking change here by privatizing  `distance`. As we’ve released v1 we could not afford that FTM.

Could you please:
- add a `distance` method that proxies call to `_distance`;
- mark it as deprecated;
- add a TODO to remind us to delete it in V2

Reply:
Done!

Reply:
Great